docs/design.md

```diff
@@ -104,3 +104,47 @@ several ways according to policy:
+
+## [Future Work] Local Attestation for Filesystem Decryption Key Provisioning
+
+![EnclaveCC_LA.png](images/EnclaveCC_LA.png)
+
+A local attestation (LA) feature is needed for authentication/authorization and secret provisioning
+between an Agent-enclave instance and a set of App-enclave instances resided in a same Pod. The
+agent holds a secret for filesystem decryption and the app wants to authenticate to the agent in
+order for the agent to release the filesystem decryption key (FDK). The app will prove its identity
+with a report and the agent will check that against a policy which requires a matching `MRSIGNER` or
+`MRENCLAVE`, possibly including an additional value from KSS. This design is a guide for enclave-CC
+developers to implement this feature.
+
+In this design, we leverages the `AEAD` algorithm for FDK protection over `ttrpc` channel. It is a
+recommended cryptographic algorithm in IETF, see [rfc5116](https://datatracker.ietf.org/doc/html/rfc5116).
+Regarding the ephemeral Diffie-Hellman key exchange, we chooses the Curve25519, which is recommended
+by IETF [rfc8031](https://www.rfc-editor.org/rfc/rfc8031). The design doesn't include the agent
+authentication in enclave-CC security context, however, the app can easily add app side user defined
+policy to enhance the security configuration.
+
+In most scenarios, it is one-shot transactions used for single filesystem decryption key (FDK)
+provisioning. However, this LA protocol can easily be extended to provision multiple keys for more
+general purposes.
+
+To achieve this, it is necessary to add a key id to the second request message sent along G_a and
+app's report, with the ID being used to indicate which key is being requested. But whether the key
+id can be automatically enumerated or not is decided by the caller `ttrpc` service/client and their
+contexts. At this moment, the caller is not discussed yet. We hope to merge the protocol definition
+completed prior to the implementation. By then, we could make the final decision to add the key id
+into protocol or not.
+
+In addition to adding this ID, the generated nonce and derived `AEAD` keyset expected to be cached
+on both sides to keep the secure session alive which has been established in initial transaction.
+The session's security context will hold those cached values to ensure AIC triad and freshness for
+subsequent key provisioning requests.
+
+Furthermore, agent or app. should set the expiration time of its own security context properly to
+minimize risk exposure. Upon closing of underlying `ttrpc` connection, the security context will be
+cleared accordingly.
+
+Please note that there is no secure transport configured between `ttrpc` endpoints in our case,
+therefore it is necessary to check the integrity and ensure the confidentiality if any of whatever
+they received using `AEAD`. Furthermore, the `ttrpc` service needs to be strictly configured to
```

Comment:
I think you just want to call out protecting the FDK specifically. If there are other messages you are trying to cover please add those or describe further what you mean. Thanks for all the updates.
```suggestion
therefore we rely on the confidentiality and integrity measures of `AEAD` to protect the FDK 
over this transport. Furthermore, the `ttrpc` service needs to be strictly configured to
```

Reply:
Thank you very much for the commit suggestion.
I think there is no other coverage,
so there is no further description for now, thanks.